### DISCUSSION TITLE
Queue Full Assert Fprime Event Logger

### DISCUSSION THREAD

**Discussion by elankford** [nasa/fprime#2825] [General]
I'm running Monte Carlo simulations with multiple instances of Fprime and am seeing the following assert randomly on a few runs. When the Fprime binary deployment is run at the start of the simulation, the assert occurs during the list of events that register variables.

Assert: /{PATH}/build-fprime-automatic-native/F-Prime/Svc/ActiveLogger/ActiveLoggerComponentAc.cpp:946 8

I tried making the EventLogger queue size bigger (10 to 100), but I am still seeing this assert sporadically. However, when I increased to 100, the assert still occurred (after 100 events instead of after 10). I'm not sure why I am seeing this Assert and why it only occurs a few times randomly during my hundreds of runs.

**Comment by LeStarch** [comment]
This is a queue overflow assert (the telltale sign is an assert with argument "8").  What this means is that you've sent more events to the active logger than it was able to process in the same time window (with an offset of `10` or `100`  of messages waiting in the queue).  Increasing the active logger component's queue size or reducing the message/event rate is the only way to fix it.

For your own components, you can use "drop on overflow" on the port to drop excess messages.  The event logger is designed to assert on overflow because a project does not wish to drop events and as such we want to see this assert while testing to ensure the queue size is large enough on launch.

**Reply by elankford** [reply]
Do you have any ideas as to why this overflow assert would occur non-deterministically (usually on 2 or 3 random runs in a set of 100)? The vast majority of runs are able to process the events properly.

**Reply by Gonta01** [reply]
@LeStarch "For your own components, you can use "drop on overflow" on the port to drop excess messages. "
How "drop on overflow" could be specified?

**Reply by zimri-leisher** [reply]
> @LeStarch "For your own components, you can use "drop on overflow" on the port to drop excess messages. " How "drop on overflow" could be specified?
```

  @ Input 2: Drop on queue full
  async input port f32ValueIn2: F32Value drop
```

See the following link for details: https://nasa.github.io/fpp/fpp-users-guide.html#Defining-Components_Port-Instances_Queue-Full-Behavior
